src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c

```diff
@@ -35,51 +42,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
+
+    Math_ApproachZeroF(&this->unk_200, 0.2f, 3.0f);
+
+    if (this->unk_166 == 1) {
+        Math_ApproachZeroF(&this->unk_1F0, 0.046f, 0.001f);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+
+        if (this->unk_164 == 0) {
+            if ((this->actor.parent != NULL) && (this->actor.parent->id == ACTOR_EN_WIZ) && (this->unk_1F0 < 0.05f)) {
+                EnWiz* wiz = (EnWiz*)this->actor.parent;
+
+                this->collider.dim.yShift = -15;
+                this->unk_164 = 1;
+                wiz->unk_3BA = 0;
+            }
+        }
+
+        if ((this->unk_164 != 0) && (this->unk_1F0 < 0.05f)) {
+            Math_ApproachZeroF(&this->unk_1FC, 1.0f, 5.0f);
+        }
+
+        if ((this->unk_1F0 < 0.001f) && (this->unk_204 < 0.001f)) {
+            D_80A4C1C0 = 0;
+            Actor_MarkForDeath(&this->actor);
+        }
+        return;
+    }
+
+    Math_ApproachZeroF(&this->unk_1F4, 0.1f, 0.01f);
+
+    if (this->unk_1F4 < 0.01f) {
+        Math_ApproachZeroF(&this->unk_1FC, 1.0f, 10.0f);
+        if ((this->unk_1FC < 10.0f) && (this->unk_204 < 0.001f)) {
+            D_80A4C1C0 = 0;
+            if ((this->actor.parent != NULL) && (this->unk_162 == 0) && (this->actor.parent->id == ACTOR_EN_WIZ) &&
+                (this->actor.parent->update != NULL)) {
+                EnWiz* wiz = (EnWiz*)this->actor.parent;
+
+                wiz->unk_3BA = 0;
+            }
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
+
+void func_80A4A608(EnWizFire* this, GlobalContext* globalCtx) {
+    if (this->unk_16A == 0) {
+        this->unk_16A = 2;
+        this->unk_16E++;
+        if (this->unk_16E >= 6) {
+            if ((this->actor.parent != NULL) && (this->unk_162 == 0) && (this->actor.parent->id == ACTOR_EN_WIZ)) {
+                EnWiz* wiz = (EnWiz*)this->actor.parent;
+
+                D_80A4C1C0 = 0;
+                if (wiz->actor.update != NULL) {
+                    wiz->unk_3BA = 0;
+                }
+            }
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
+
+void EnWizFire_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    static Color_RGB8 D_80A4C234[] = {
+        { 100, 40, 40 }, { 180, 0x78, 80 }, { 155, 80, 80 },   { 125, 20, 0 },
+        { 0, 0, 0 },     { 200, 250, 250 }, { 100, 250, 250 }, { 225, 255, 235 },
+    };
+    static Color_RGBA8 D_80A4C24C = { 250, 250, 250, 255 };
+    static Color_RGBA8 D_80A4C250 = { 180, 180, 180, 255 };
+    GlobalContext* globalCtx = globalCtx2;
+    EnWizFire* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+    s32 j;
+    s16 temp_s0;
+    s16 idx;
+
+    Actor_SetScale(&this->actor, this->unk_14C);
+    func_80A4BDDC(this, globalCtx);
+    this->unk_204 = this->unk_200 / 60.0f;
+
+    if (this->unk_162 == 0) {
+        EnWiz* wiz = (EnWiz*)this->actor.parent;
+
+        if ((wiz != NULL) && (wiz->actor.id == ACTOR_EN_WIZ) && (wiz->actor.update != NULL) && (wiz->unk_74A != 2)) {
```

Comment:
Kind of minor, but I would prefer this variable just being type Actor, then cast it to `EnWiz` when checking `unk_74A`. As an example of something similar see `func_80BD9234` in `z_en_pamera`.